File: src/player.py
class Player:
    def __init__(self, name):
        self.name = name
        self.wins = 0
        self.points = 0

    def add_win(self):
        self.wins = self.wins + 1

    def add_winnerscore(self, score):
        self.points = self.points + (21 - score)

    def add_looserscore(self, score):
        self.points = self.points - (21 - score)

    def add_player(self, name):
        self.name.append(name)

    def print_standings(self):
        print(f"Navn: {self.name}")
        print(f"Wins: {self.wins}")
        print(f"Points: {self.points}")

    def print_name(self):
        print(f"Navn: {self.name}")



    

File: docs/js/player.js
// Mirrors src/player.py's Player class. Plain data + pure functions (not a class)
// so player objects round-trip through JSON/localStorage without losing behavior.

let nextId = 1;

export function createPlayer(name) {
  return {
    id: "p" + (nextId++) + "-" + Date.now().toString(36),
    name,
    wins: 0,
    points: 0,
    streak: 0,
    change: 0,
    satOutLastRound: false,
    walkoverScore: 0, // gradient-weighted walkover credit -- see buildRound in tournament.js
    history: [], // { round, type: "match"|"walkover", teammate, opponents, myScore, opponentScore, result }
    pendingRemoval: false, // "remove from next round" (admin table) -- see confirmRoundEnd in app.js
  };
}

// player.py: add_win()
export function addWin(player) {
  player.wins += 1;
}

// player.py: add_winnerscore(score) -- score is the LOSING team's score
export function addWinnerScore(player, loserScore) {
  player.points += 21 - loserScore;
}

// player.py: add_looserscore(score) -- score is the LOSING team's own score
export function addLooserScore(player, loserScore) {
  player.points -= 21 - loserScore;
}
